azure-iot-device/azure/iot/device/iothub/aio/async_handler_manager.py

```diff
@@ -88,36 +66,138 @@ def _handler_callback(future):
                 fut = tpe.submit(handler, handler_arg)
                 fut.add_done_callback(_handler_callback)
 
-    async def _event_handler_runner(self, handler_name):
-        # TODO: implement
-        logger.error("._event_handler_runner() not yet implemented")
-
-    def _start_handler_runner(self, handler_name):
-        """Create, and store a task for running a handler
+    async def _client_event_handler_runner(self):
```

Comment:
>_client_event_handler_runner [](http://example.com/codeflow?start=14&length=28)

Not a comment on the quality of your code or this change in particular, but I can't get over the thought that maybe we over-designed this part of the code (the handler managers and runners).  We did have legacy receive APIs to contend with, but I feel like this _should_ be simpler.  I couldn't say how though  It's just a gut feel.  (No change requested.  Just pontificating.)

Reply:
I agree - there's some things we might be able to improve in the future that I have ideas on as well.